Objects/obmalloc.c

```diff
@@ -1562,8 +1562,14 @@ _PyObject_DebugReallocApi(char api, void *p, size_t nbytes)
      * but we live with that.
      */
     q = (uchar *)PyObject_Realloc(q - 2*SST, total);
-    if (q == NULL)
+    if (q == NULL) {
+        if (nbytes < original_nbytes) {
```

Comment:
Should this be `<=`? I know it's checking the memset logic above, but I think we assume the "shrinking" to the same size always works, too.

Reply:
Right. I changed the code for <=